src/measure_theory/lebesgue_measure.lean

```diff
@@ -378,6 +378,74 @@ begin
   simpa [-mem_Ioo] using hx.1.trans hx.2
 end
 
+section region_between
+
+open_locale classical
+
+variable {α : Type*}
+
+/-- The region between two real-valued functions on an arbitrary set. -/
+def region_between (f g : α → ℝ) (s : set α) : set (α × ℝ) :=
+{ p : α × ℝ | p.1 ∈ s ∧ p.2 ∈ Ioo (f p.1) (g p.1) }
+
+variables [measurable_space α] {μ : measure α} {f g : α → ℝ} {s : set α}
+
+/-- The region between two measurable functions on a measurable set is measurable. -/
+lemma is_measurable_region_between
+  (hf : measurable f) (hg: measurable g) (hs : is_measurable s) :
+  is_measurable (region_between f g s) :=
+begin
+  dsimp only [region_between, Ioo, mem_set_of_eq, set_of_and],
+  refine is_measurable.inter _ ((is_measurable_lt (hf.comp measurable_fst) measurable_snd).inter
+    (is_measurable_lt measurable_snd (hg.comp measurable_fst))),
+  convert hs.prod is_measurable.univ,
+  simp only [and_true, mem_univ],
+end
+
+/-- The volume of the region between two measurable functions on a measurable set can be
+    respresented as a Lebesgue integral. -/
+theorem volume_region_between_eq_lintegral
```

Comment:
Can you deduce from this one a version where you only assume almost everywhere measurability of `f` and `g` with respect to `μ.restrict s` instead of measurability? It should be possible to deduce it from the current version applied to measurable versions of `f` and `g` (constructed with `ae_measurable.mk`). Then in the next theorem you can remove the measurability assumptions (as integrable implies almost everywhere measurable by definition).

Reply:
Ah, this is interesting. We realized we probably ought to weaken some assumptions in the integral theorems. Last week, I tried to prove an alternate version of `is_measurable_region_between` that only took `ae_measurable` as assumptions instead of `measurable`. That turned out to be tricky (and I'm still not sure if that proof is possible). 

I'll work on proving this alternate version of `eq_lintegral`.  It looks more promising. Just curious, do you think that original idea about `is_measurable_region_between` would actually  be possible? Or do we have to keep this work "inside of" `eq_lintegral` like you suggest?

Reply:
If the functions are only almost everywhere measurable, then the region between their graphs will not be measurable in general -- it will only be equal to a measurable set up to a set of measure zero. So using almost measurable only for the integral version sounds reasonable.

Reply:
I just pushed the changes you requested! I added new versions of `eq_lintegral` and `eq_integral` with fewer assumptions. I also added a few small lemmas in measure_theory.measure_space and measure_theory.prod that I had to prove along the way.